docs/SILProgrammersManual.md

```diff
@@ -159,6 +159,165 @@ rules out PartialApplies is no excuse to conflate applied arguments
 with function arguments. Also, without consistent use of common
 idioms, it becomes overly burdensome to evolve these APIs over time.
 
+## `AccessedStorage` and `AccessPath`
+
+The `AccessedStorage` and `AccessPath` types formalize memory access
+in SIL. Given an address-typed SIL value, it must be possible to
+reliably identify the storage location of the accessed
+memory. `AccessedStorage` identifies an accessed storage location. It
+is a value object that requires no memory allocation and can be used
+as the hash key for that location. Extra bits are also available for
+information specific to a particular optimization pass. Its API
+provides the kind of location being accessed and information about the
+location's uniqueness or whether it is distinct from other storage.
+
+### Uniquely identified storage
+
+Two uniquely identified storage locations may only alias if their
+AccessedStorage objects are identical.
+
+`AccessedStorage::isUniquelyIdentifiedStorage()` is a weaker check that
+does not depend on exclusivity enforcement.
+
+`AccessedStorage::isUniquelyIdentifiedBase()` is a stronger check that
+depends on exclusivity enforcement.
```

Comment:
I merged two separate `isUniquelyIdentified` APIs and wanted to preserve their functionality, at least in this PR. But that was too confusing. Since then, I spent time proving to myself that our existing optimizations will all still be correct if we make aggressive assumptions everywhere.